### DISCUSSION TITLE
How to use MATLAB wrap to package custom factors

### DISCUSSION THREAD

**Discussion by zhaoxin0716** [borglab/gtsam#1515] [Q&A] [answered]
Hello everyone, I would like to ask if I have customized a factor through gtsam . How can I use the MATLAB wrapper to package it? Thank you

**Comment by mikesheffler** [comment, marked as answer]
It's a pretty old reference now, but Jing Dong put together [a talk](https://dongjing3309.github.io/files/gtsam-tutorial.pdf) that includes a section on making custom factors in MATLAB. The slides aren't numbered, but look about halfway down, where the titles are `Write your own factor` and then `Use your own factor in MATLAB`. There's a link to a [repo](https://github.com/dongjing3309/gtsam-examples) where he has code for the examples.

That talk is like six years old, but I used it to make a custom factor in MATLAB maybe three, four years ago?

There could be changes in the MATLAB wrap that make this easier to do now, but I wouldn't know -- I mostly use the MATLAB toolbox as-is for plotting stuff I did in C++.

Does anyone have a modern example showing how to do a custom factor for use in MATLAB? I'm not sure who the MATLAB gurus are now. Maybe @mattking-smith ?
